tensorpipe/channel/cuda_ipc/channel.cc

```diff
@@ -77,9 +79,26 @@ class SendOperation {
   Descriptor descriptor() {
     cudaIpcMemHandle_t handle;
     TP_CUDA_CHECK(cudaIpcGetMemHandle(&handle, const_cast<void*>(ptr_)));
-
+    CUdeviceptr basePtr;
+
+    CUresult error = cuMemGetAddressRange(
+        &basePtr, nullptr, reinterpret_cast<CUdeviceptr>(ptr_));
+    if (error != CUDA_SUCCESS) {
+      CUresult res;
+      const char* errorName;
+      const char* errorStr;
+      res = cuGetErrorName(error, &errorName);
+      TP_THROW_ASSERT_IF(res != CUDA_SUCCESS);
+      res = cuGetErrorString(error, &errorStr);
+      TP_THROW_ASSERT_IF(res != CUDA_SUCCESS);
+
+      TP_THROW_ASSERT() << "CUDA error (" << errorName << "): " << errorStr;
+    }
```

Comment:
This could be extracted to a helper function/macro, but since we're using the driver API only here for now I left it inline.